openssh/client/service.sls

```diff
@@ -36,6 +36,7 @@ openssh_client_config:
   - makedirs: true
   - require:
     - pkg: openssh_client_packages
+    - file: {{ user.user.home }}
```

Comment:
I don't think it belongs here. There's no include of linux states out of openssh.client.service which provides this file resource. So it will most likely fail without running together with linux state.

This is common issue - eg. formula B assumes that formula A was executed successfuly in order to complete it's job but it doesn't enforce it's execution by depending on formula A's states and resources.
Solution is to ensure states will be executed in correct order - for example by using Salt Orchestrate.

Reply:
Ok, I've never used salt orchestrate: is this a new way to order states?

Reply:
No, it's pretty long.
There was a WIP to add support to generate orchestrate files, see https://github.com/salt-formulas/salt-formula-openssh/blob/master/openssh/meta/salt.yml#L1
but AFAIK never finished. But you can create orchestrate files on your own.

Reply:
naive question: as openssh formula actually depends on linux formula, why not introduce cross-formula "require" dependencies, forcing to run the openssh.client state together with linux state, that would sound sensible to me, no?

Reply:
And what about FreeBSD, OS X and others? :-)

Reply:
I understand.